/*
 * This file was automatically generated by EvoSuite
 * Sat Jun 22 14:25:32 GMT 2024
 */

package org.apache.commons.fileupload;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import java.io.File;
import java.io.PrintStream;
import java.io.PrintWriter;
import org.apache.commons.fileupload.FileUploadException;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.evosuite.runtime.mock.java.io.MockFile;
import org.evosuite.runtime.mock.java.io.MockPrintStream;
import org.evosuite.runtime.mock.java.io.MockPrintWriter;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true) 
public class FileUploadException_ESTest extends FileUploadException_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      FileUploadException fileUploadException0 = FileUploadException.FileUploadException0();
      FileUploadException fileUploadException1 = new FileUploadException("Caused by:", fileUploadException0);
      Throwable throwable0 = fileUploadException1.getCause();
      assertNotSame(fileUploadException1, throwable0);
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      FileUploadException fileUploadException0 = FileUploadException.FileUploadException0();
      // Undeclared exception!
      try { 
        fileUploadException0.printStackTrace1((PrintWriter) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.evosuite.runtime.mock.java.lang.MockThrowable", e);
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      FileUploadException fileUploadException0 = FileUploadException.FileUploadException0();
      // Undeclared exception!
      try { 
        fileUploadException0.printStackTrace0((PrintStream) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.evosuite.runtime.mock.java.lang.MockThrowable", e);
      }
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      FileUploadException fileUploadException0 = FileUploadException.FileUploadException0();
      File file0 = MockFile.createTempFile("Caused by:", "Caused by:");
      MockPrintWriter mockPrintWriter0 = new MockPrintWriter(file0);
      FileUploadException fileUploadException1 = new FileUploadException("i[O", fileUploadException0);
      fileUploadException1.printStackTrace1(mockPrintWriter0);
      assertFalse(fileUploadException1.equals((Object)fileUploadException0));
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      FileUploadException fileUploadException0 = FileUploadException.FileUploadException0();
      MockFile mockFile0 = new MockFile("~g$9\"Ph>8(OL");
      MockPrintStream mockPrintStream0 = new MockPrintStream(mockFile0);
      MockPrintWriter mockPrintWriter0 = new MockPrintWriter(mockPrintStream0);
      fileUploadException0.printStackTrace1(mockPrintWriter0);
  }

  @Test(timeout = 4000)
  public void test5()  throws Throwable  {
      FileUploadException fileUploadException0 = FileUploadException.FileUploadException0();
      FileUploadException fileUploadException1 = new FileUploadException("Caused by:", fileUploadException0);
      MockFile mockFile0 = new MockFile("~g$9\"Ph>8(OL");
      MockPrintStream mockPrintStream0 = new MockPrintStream(mockFile0);
      fileUploadException1.printStackTrace0(mockPrintStream0);
      //  // Unstable assertion: assertEquals(438L, mockFile0.length());
  }

  @Test(timeout = 4000)
  public void test6()  throws Throwable  {
      FileUploadException fileUploadException0 = FileUploadException.FileUploadException0();
      MockPrintStream mockPrintStream0 = new MockPrintStream("org.apache.commons.fileupload.FileUploadException");
      fileUploadException0.printStackTrace0(mockPrintStream0);
  }

  @Test(timeout = 4000)
  public void test7()  throws Throwable  {
      FileUploadException fileUploadException0 = FileUploadException.FileUploadException1("Caused by:");
      assertNotNull(fileUploadException0);
  }

  @Test(timeout = 4000)
  public void test8()  throws Throwable  {
      FileUploadException fileUploadException0 = FileUploadException.FileUploadException0();
      Throwable throwable0 = fileUploadException0.getCause();
      assertNull(throwable0);
  }
}
